### PR TITLE
CI for e2e testing 

### PR DESCRIPTION
**Pull request by shapirov103**
*Issue #, if available:* #31

*Description of changes:* CI based on codebuild integration to run e2e. Issue updated with the implementation and configuration.  See https://github.com/aws-observability/cdk-aws-observability-accelerator/issues/31#issuecomment-1652258284 for more info.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by shapirov103** [comment]
@elamaran11 the current implementation of the GitHub action requires access to the GitHub token (ideally of a technical user like CIBot) for integration between CodeBuild and the PR under review. I kept the integration somewhat similar to what we have on the Blueprints side. BTW, I will move it over to the blueprints as well so that core contributors could kick off e2e tests.

File: .github/workflows/e2e.yaml
Comment (elamaran11) [inline .github/workflows/e2e.yaml:34]: Lets remove these comments?

File: ci/buildspec.yml
Comment (elamaran11) [inline ci/buildspec.yml:30]: Lets remove comments.

File: Makefile
Comment (elamaran11) [inline Makefile:62]: Why is this required and where is this user?

File: ci/buildspec.yml
Comment (elamaran11) [inline ci/buildspec.yml:1]: Where is `buildspec.yml` used?

**Review by elamaran11 — COMMENTED** [review]
@shapirov103 Awesome work and thankyou for doing this. We need following update as discussed
1. Allow only admin users of the repo group to run the `e2e`
2. We have to work on creating a Bot User (I can take it)

File: Makefile
Comment (shapirov103) [reply Makefile:62]: This is checking if target account was bootstrapped with CDK (`cdk bootstrap`)

**Comment by shapirov103** [comment]
@elamaran11 looks like the repo is broken now, the failures above are not related to my changes. 

Because the build is broken I was not able to validate if codebuild works fully, however I tested the permission check and it is currently testing for `write` and `contributor`. 

If merged, we can test with `/do-e2e-test single-new-eks-cluster deploy` and `/do-e2e-test single-new-eks-cluster destroy` commands.

**Comment by elamaran11** [comment]
> @elamaran11 looks like the repo is broken now, the failures above are not related to my changes.
> 
> Because the build is broken I was not able to validate if codebuild works fully, however I tested the permission check and it is currently testing for `write` and `contributor`.
> 
> If merged, we can test with `/do-e2e-test single-new-eks-cluster deploy` and `/do-e2e-test single-new-eks-cluster destroy` commands.

@shapirov103 The repo was broken because of breaking change to FluxAddon which we should have prepared ahead. Now the broken change is fixed. Please proceed with testing again and the failures will go away. Let me know once GH action passes, we can merge this to test.

File: ci/buildspec.yml
Comment (elamaran11) [reply ci/buildspec.yml:1]: Got it.

File: .github/workflows/e2e.yaml
Comment (shapirov103) [reply .github/workflows/e2e.yaml:46]: they are not part of the execution but if you ever need to override the buildspec for codebuild this will be handy.

File: ci/buildspec.yml
Comment (shapirov103) [reply ci/buildspec.yml:30]: I kept if we decide we need to clean up always

**Review by elamaran11 — APPROVED** [review]
LGTM

**Pull request merged**
